CollAction/Services/Project/ProjectService.cs

```diff
@@ -285,62 +249,107 @@ private string GetParticipantCsvLine(ApplicationUser user)
+                ? CreateFindProjectsViewModel(project)
+                : null;
+        }
+
+        public IQueryable<FindProjectsViewModel> FindProjects(Expression<Func<Models.Project, bool>> filter, int? limit, int? start)
+        {
+            IQueryable<Models.Project> projects = _context
+                .Projects
+                .Include(p => p.Category)
+                .Include(p => p.Location)
+                .Include(p => p.BannerImage)
+                .Include(p => p.ParticipantCounts)
+                .Where(filter)
+                .OrderBy(p => p.DisplayPriority);
+
+            if (limit.HasValue)
+                projects = projects.Take(limit.Value);
+
+            return projects.Select(p => CreateFindProjectsViewModel(p));
+        }
         public async Task<IEnumerable<FindProjectsViewModel>> MyProjects(string userId)
         {
-            return await _context.Projects
+            var projects = await _context
+                .Projects
                 .Where(p => p.OwnerId == userId && p.Status != ProjectStatus.Deleted)
+                .Include(p => p.Category)
+                .Include(p => p.Location)
+                .Include(p => p.BannerImage)
+                .Include(p => p.ParticipantCounts)                
                 .OrderBy(p => p.DisplayPriority)
-                .Select(project => new FindProjectsViewModel(_stringLocalizer) // Cannot be moved to private method. Won't work with EF
+                .ToListAsync();
+
+            var viewModels = projects.Select(p => 
                 {
-                    ProjectId = project.Id,
-                    ProjectName = project.Name,
-                    ProjectNameUriPart = GetProjectNameNormalized(project.Name),
-                    ProjectProposal = project.Proposal,
-                    CategoryName = project.Category.Name,
-                    CategoryColorHex = project.Category.ColorHex,
-                    LocationName = project.Location.Name,
-                    BannerImagePath = project.BannerImage != null ? _imageService.GetUrl(project.BannerImage) : $"/images/default_banners/{project.Category.Name}.jpg",
-                    BannerImageDescription = project.BannerImage.Description,
-                    Target = project.Target,
-                    Participants = project.ParticipantCounts.Count,
-                    Remaining = project.RemainingTime,
-                    DescriptiveImagePath = project.DescriptiveImage == null ? null : _imageService.GetUrl(project.DescriptiveImage),
-                    DescriptiveImageDescription = project.DescriptiveImage.Description,
-                    Status = project.Status,
-                    Start = project.Start,
-                    End = project.End,
-                    CanSendProjectEmail = this.CanSendProjectEmail(project)
+                    var viewModel = CreateFindProjectsViewModel(p);
+                    viewModel.CanSendProjectEmail = this.CanSendProjectEmail(p);
```

Comment:
Bit unrelated, but CanSendProjectEmail looks like it might be better as an Unmapped property on the project model.

Reply:
It uses two injected configuration settings, " _projectEmailOptions.MaxNumberProjectEmails" and "_projectEmailOptions.TimeEmailAllowedAfterProjectEnd" so it can't be a derived property.